sdk/textanalytics/ai-text-analytics/src/transforms.ts

```diff
@@ -0,0 +1,346 @@
+  };
+}
+
+/**
+ * Converts a target relation object returned by the service to an assessment
+ * sentiment object where JSON pointers in the former are realized in the
+ * latter.
+ *
+ * @param targetRelation - The target relation object to be converted.
+ * @param response - The entire response returned by the service.
+ * @returns The user-friendly assessment sentiment object.
+ * @internal
+ */
+function convertTargetRelationToAssessmentSentiment(
+  targetRelation: TargetRelation,
+  sentences: GeneratedSentenceSentiment[]
+): AssessmentSentiment {
+  const assessmentPtr = targetRelation.ref;
+  const assessmentIndex: AssessmentIndex = parseAssessmentIndex(assessmentPtr);
+  const assessment: SentenceAssessment | undefined =
+    sentences?.[assessmentIndex.sentence].assessments?.[assessmentIndex.assessment];
+  if (assessment !== undefined) {
+    return assessment;
+  } else {
+    throw new Error(`Pointer "${assessmentPtr}" is not a valid Assessment pointer`);
+  }
+}
+
+/**
+ * @internal
+ */
+export function toEntityLinkingResult(
+  documents: TextDocumentInput[],
+  results: GeneratedEntityLinkingResult
+): EntityLinkingResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function toKeyPhraseExtractionResult(
+  documents: TextDocumentInput[],
+  results: GeneratedKeyPhraseExtractionResult
+): KeyPhraseExtractionResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function toEntityRecognitionResult(
+  documents: TextDocumentInput[],
+  results: GeneratedEntityRecognitionResult
+): EntityRecognitionResult[] {
+  return transformDocumentResults(documents, results);
+}
+
+/**
+ * @internal
+ */
+export function transformActionResult(
+  actionName: AnalyzeActionName,
+  input: TextDocumentInput[] | LanguageDetectionInput[],
+  response: AnalyzeResponse
+):
+  | EntityLinkingResult[]
+  | EntityRecognitionResult[]
+  | KeyPhraseExtractionResult[]
+  | SentimentAnalysisResult[]
+  | LanguageDetectionResult[] {
+  switch (response.kind) {
+    case "EntityLinkingResults": {
+      return toEntityLinkingResult(input, (response as EntityLinkingTaskResult).results);
+    }
+    case "EntityRecognitionResults": {
+      return toEntityRecognitionResult(input, (response as EntitiesTaskResult).results);
+    }
+    case "KeyPhraseExtractionResults": {
+      return toKeyPhraseExtractionResult(input, (response as KeyPhraseTaskResult).results);
+    }
+    case "PiiEntityRecognitionResults": {
+      return toPiiEntityRecognitionResult(input, (response as PiiTaskResult).results);
+    }
+    case "SentimentAnalysisResults": {
+      return toSentimentAnalysisResult(input, (response as SentimentTaskResult).results);
+    }
+    case "LanguageDetectionResults": {
+      return toLanguageDetectionResult(input, (response as LanguageDetectionTaskResult).results);
+    }
+    default: {
+      throw new Error(`Unsupported action name: ${actionName}`);
```

Comment:
At the bottom of exhaustive `switch`es, I like to add an exhaustiveness check if it can be supported.

Also, the error message references a different value than the `switch`.

```suggestion
      const __exhaust: never = response;
      throw new Error(`Unsupported response type: ${__exhaust}`);
```

Reply:
Love it, thanks!